src/main/java/org/folio/gobi/LookupService.java

```diff
@@ -68,19 +69,26 @@ public LookupService(RestClient restClient) {
    * @param location
    * @return UUID of the location
    */
-  public CompletableFuture<String> lookupLocationId(String location) {
+  public CompletableFuture<LocationTranslationResult> lookupLocationId(String location) {
     logger.debug("lookupLocationId:: Trying to look up locationId by location '{}'", location);
-    String query = HelperUtils.encodeValue(String.format(CQL_CODE_STRING_FMT, location));
-    String endpoint = String.format(LOCATIONS_ENDPOINT + QUERY, query);
-    return restClient.handleGetRequest(endpoint).toCompletionStage().toCompletableFuture()
+    return restClient.handleGetRequest(LOCATIONS_ENDPOINT + "?limit=1000").toCompletionStage().toCompletableFuture()
       .thenCompose(locations -> {
-        String locationId = HelperUtils.extractLocationId(locations);
-        if (StringUtils.isEmpty(locationId)) {
-          logger.warn("lookupLocationId:: Location '{}' not found", location);
-          return completedFuture(null);
+        JsonArray jsonArray = locations.getJsonArray("locations");
+        for (int i = 0; i < jsonArray.size(); i++) {
+          JsonObject locationObject = jsonArray.getJsonObject(i);
+          if (location != null && location.equals(locationObject.getString("code"))) {
```

Comment:
let's think about two edge cases here.
1. What is location == null. Seems there is no need to work when location null, so we can do return  `return completedFuture(null);` in the beginning before send request to mod-search.
2. What is jsonArray is empty. then we can't get locationObject. we can also do condition for this

Reply:
1. Location will not be null when called from FE. With this check just avoided NPE when working from swagger.
2. For empty jsonArray for cycle won't be executed so code will return correct result - null.

Reply:
That is make sense. I got it